src/main/java/uk/ac/ebi/eva/pipeline/io/writers/VariantMongoWriter.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2016 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.pipeline.io.writers;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.BulkWriteOperation;
+import com.mongodb.DBCollection;
+import com.mongodb.DBObject;
+import org.opencb.biodata.models.variant.Variant;
+import org.opencb.biodata.models.variant.VariantSourceEntry;
+import org.opencb.opencga.storage.mongodb.variant.DBObjectToVariantConverter;
+import org.opencb.opencga.storage.mongodb.variant.DBObjectToVariantSourceEntryConverter;
+import org.opencb.opencga.storage.mongodb.variant.DBObjectToVariantStatsConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.data.MongoItemWriter;
+
+import java.util.List;
+
+/**
+ * @author Diego Poggioli
+ *
+ * Write a list of {@link Variant} into MongoDB
+ * See also {@link org.opencb.opencga.storage.mongodb.variant.VariantMongoDBWriter}
+ *
+ */
+public class VariantMongoWriter extends MongoItemWriter<Variant> {
+    private static final Logger logger = LoggerFactory.getLogger(VariantMongoWriter.class);
+
+    private boolean includeStats;
+    private String fileId;
+    private int bulkSize = 0;
+    private DBCollection variantsCollection;
+
+    private BulkWriteOperation bulk;
+    private long numVariantsWritten;
+    private int currentBulkSize = 0;
+
+    private DBObjectToVariantConverter variantConverter;
+    private DBObjectToVariantStatsConverter statsConverter;
+    private DBObjectToVariantSourceEntryConverter sourceEntryConverter;
+
+    public VariantMongoWriter(boolean includeStats, String fileId, int bulkSize,
+                              DBObjectToVariantConverter variantConverter, DBObjectToVariantStatsConverter statsConverter,
+                              DBObjectToVariantSourceEntryConverter sourceEntryConverter, DBCollection variantsCollection) {
```

Comment:
This is quite different from [how the `VepAnnotationMongoWriter` was implemented](https://github.com/EBIvariation/eva-pipeline/blob/develop/src/main/java/uk/ac/ebi/eva/pipeline/io/writers/VepAnnotationMongoWriter.java).

A `MongoOperations` object would define how to connect to the database. It supports bulk operations so it shouldn't be a problem. A `Converter` would wrap the OpenCGA ones.